File: game-server/src/handlers/battle/pve_battle_result.rs
use sr_proto::prost::Message;
use sr_proto::{PveBattleResultCsReq, PveBattleResultScRsp};

pub async fn handle(req: &[u8]) -> PveBattleResultScRsp {
    let req = PveBattleResultCsReq::decode(req).unwrap_or_default();

    PveBattleResultScRsp {
        retcode: 0,
        end_status: req.end_status,
        stage_id: req.stage_id,
        battle_id: req.battle_id,
        ..Default::default()
    }
}

File: gameserver/handler/battle/PveBattleResult.go
package battle

import (
	"github.com/yuvlian/qingque-sr/gameserver/session"
	"github.com/yuvlian/qingque-sr/pb"
	"github.com/yuvlian/qingque-sr/pb/cid"
)

func PveBattleResult(s *session.Session) error {
	var req pb.PVEBattleResultCsReq
	if err := s.DecodeReq(&req); err != nil {
		return err
	}

	rsp := &pb.PVEBattleResultScRsp{
		EndStatus: req.EndStatus,
		StageId:   req.StageId,
		BattleId:  req.BattleId,
	}
	return s.Send(cid.PVEBattleResultScRsp, rsp)
}
